include/boost/graph/subgraph.hpp

```diff
@@ -131,29 +131,35 @@ class subgraph {
 
     // copy constructor
     subgraph(const subgraph& x)
-        : m_parent(x.m_parent), m_edge_counter(x.m_edge_counter)
-        , m_global_vertex(x.m_global_vertex), m_global_edge(x.m_global_edge)
+        : m_parent(x.m_parent), m_edge_counter(0)
     {
         if(x.is_root())
         {
-         m_graph = x.m_graph;
+            m_graph = x.m_graph;
+            m_edge_counter = x.m_edge_counter;
```

Comment:
If the subgraph is actually the entire graph keep count of its number of edges, else don't keep track of the member graph's number of edges?  This seems like a change in function.

Reply:
If the `m_graph` is a child of its parent, we manually add all the vertices from `x.m_graph` to `m_graph`. This will automatically increase the `m_edge_counter` instead of assign `x.m_edge_counter` to it.